src/composables/useModelMigration.ts

```diff
@@ -0,0 +1,32 @@
+/**
+ * SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+import { getCurrentInstance, computed } from 'vue'
+
+/**
+ * Create model proxy to new v9 model (modelValue + update:modelValue) with a fallback to old model
+ * @param {string} oldModelName - Name of model prop in nextcloud-vue v8
+ * @param {string} oldModelEvent - Event name of model event in nextcloud-vue v8
+ * @return {import('vue').WritableComputedRef} - model proxy
+ */
+export function useModelMigration(oldModelName, oldModelEvent) {
+    const vm = getCurrentInstance()!.proxy
```

Comment:
I am not sure here, I had issues with this in past (accessing the `proxy` inside of the setup function).
Not sure if it was something in my code but for me I had to move the `proxy` access into the computed functions

Reply:
We've been using similar approach for several major versions already in Talk for Vuex Store (useStore.js) in script setup components and composables, and so far didn't caught any errors with it.
Maybe some guard here with debug information would be safe (so if anyone encounters it, we can identify the reason)

Reply:
@susnux This is not possible. This function returns either `null` or `{ proxy: instance }`.

https://github.com/vuejs/vue/blob/main/src/v3/currentInstance.ts#L13